core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -380,13 +386,19 @@ private BasicInvocationStage handleNonTxWriteCommand(InvocationContext ctx, Writ
          }
          if (!(ce instanceof OutdatedTopologyException) && !(ce instanceof SuspectException))
             throw t;
+         WriteCommand writeCommand = (WriteCommand) rCommand;
+         if (writeCommand.hasFlag(Flag.ZERO_LOCK_ACQUISITION_TIMEOUT)) {
+            // the command is expected to not wait
```

Comment:
You could also say that JGroups also uses locking under the hood (e.g. in UFC/MFC), so you shouldn't do any RPCs with `ZERO_LOCK_ACQUISITION_TIMEOUT`. But obviously we want to to RPCs for `putForExternalRead`, so we have to limit the semantics of ZLAT at some point.